src/contrib/hybrid/codegen_hybrid.cc

```diff
@@ -0,0 +1,490 @@
+    PrintBinaryIntrinsitc(op, "&", os, this);
+  } else if (op->is_intrinsic(Call::bitwise_xor)) {
+    PrintBinaryIntrinsitc(op, "^", os, this);
+  } else if (op->is_intrinsic(Call::bitwise_or)) {
+    PrintBinaryIntrinsitc(op, "|", os, this);
+  } else if (op->is_intrinsic(Call::shift_left)) {
+    PrintBinaryIntrinsitc(op, "<<", os, this);
+  } else if (op->is_intrinsic(Call::shift_right)) {
+    PrintBinaryIntrinsitc(op, ">>", os, this);
+  } else if (op->is_intrinsic(Call::bitwise_not)) {
+    CHECK_EQ(op->args.size(), 1U);
+    os << "(~";
+    PrintExpr(op->args[0], os);
+    os << ')';
+  } else if (op->is_intrinsic(intrinsic::tvm_if_then_else)) {
+    PrintExpr(op->args[1], os);
+    os << " if ";
+    PrintExpr(op->args[0], os);
+    os << " else ";
+    PrintExpr(op->args[2], os);
+  } else {
+    os << op->name << "(";
+    for (size_t i = 0; i < op->args.size(); i++) {
+      PrintExpr(op->args[i], os);
+      if (i < op->args.size() - 1) {
+        os << ", ";
+      }
+    }
+    os << ")";
+  }
+}
+
+void CodeGenHybrid::VisitExpr_(const Load* op, std::ostream& os) {  // NOLINT(*)
+  LOG(FATAL) << "Phase 0 has no Load(s)!";
+}
+
+void CodeGenHybrid::VisitStmt_(const Store* op) {
+  LOG(FATAL) << "Phase 0 has no Store(s)!";
+}
+
+void CodeGenHybrid::VisitExpr_(const Let* op, std::ostream& os) {  // NOLINT(*)
+  LOG(FATAL) << "Phase 0 has no Let(s)!";
+}
+
+void CodeGenHybrid::VisitStmt_(const Allocate* op) {
+  LOG(FATAL) << "Phase 0 has no Allocate(s)!";
+}
+
+void CodeGenHybrid::VisitExpr_(const Ramp* op, std::ostream& os) {  // NOLINT(*)
+  LOG(FATAL) << "Ramp to be supported yet";
+}
+
+void CodeGenHybrid::VisitExpr_(const Broadcast* op, std::ostream& os) {   // NOLINT(*)
+  LOG(FATAL) << "Broadcast: not supported ";
+}
+
+void CodeGenHybrid::VisitExpr_(const Select* op, std::ostream& os) {  // NOLINT(*)
+  PrintExpr(op->true_value, os);
+  os << " if ";
+  PrintExpr(op->condition, os);
+  os << " else ";
+  PrintExpr(op->false_value, os);
+  os << "\n";
+}
+
+void CodeGenHybrid::VisitStmt_(const LetStmt* op) {
+  std::string value = PrintExpr(op->value);
+  stream << GetVarID(op->var.get()) << " = " << value << ";\n";
+  PrintStmt(op->body);
+}
+
+void CodeGenHybrid::VisitStmt_(const AttrStmt* op) {
+  if (op->attr_key == ir::attr::thread_extent) {
+    auto iter_var = op->node.as<IterVarNode>();
+    CHECK(iter_var);
+    binds_[iter_var->var.get()] = dot_to_underscore(iter_var->var->name_hint);
+    PrintIndent();
+    stream << "for " << binds_[iter_var->var.get()] << " in bind('"
+           << iter_var->var->name_hint << "', ";
+    PrintExpr(op->value, stream);
+    stream << "):\n";
+    indent_ += tab_;
+    PrintStmt(op->body);
+    indent_ -= tab_;
+  } else if (op->attr_key == ir::attr::realize_scope) {
+    auto v = FunctionRef(op->node.node_);
+    alloc_storage_scope_[v] = op->value.as<StringImm>()->value;
+    PrintStmt(op->body);
+  } else {
+    // For now we ignore the unsupported AttrStmt
```

Comment:
what about printing some warnings?

Reply:
I would rather not do that. For now, the AttrStmt, which is both not supported and will be faced, is only buffer binding hints. This will be resolved by later stages.